src/applications/disability-benefits/all-claims/tests/utils/dates/partial-date-validations.unit.spec.jsx

```diff
@@ -0,0 +1,277 @@
+      validateRequiredPartialDate(
+        errorsSpy,
+        { month: '1', day: '15', year: '2023' },
+        {},
+        {},
+      );
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+  });
+
+  describe('validatePartialDateRange', () => {
+    it('should validate valid date ranges', () => {
+      validatePartialDateRange(errorsSpy, '2023-01-01', '2023-12-31');
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+
+    it('should validate partial date ranges', () => {
+      validatePartialDateRange(errorsSpy, '2023-01-XX', '2023-12-XX');
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+
+    it('should validate year-only ranges', () => {
+      validatePartialDateRange(errorsSpy, '2020-XX-XX', '2023-XX-XX');
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+
+    it('should error when from date is invalid', () => {
+      validatePartialDateRange(errorsSpy, 'invalid-from-date', '2023-12-31');
+
+      expect(errorsSpy.addError.calledOnce).to.be.true;
+    });
+
+    it('should error when to date is invalid', () => {
+      validatePartialDateRange(errorsSpy, '2023-01-01', 'invalid-to-date');
+
+      expect(errorsSpy.addError.calledOnce).to.be.true;
+    });
+
+    it('should error when end date is before start date', () => {
+      validatePartialDateRange(errorsSpy, '2023-12-31', '2023-01-01');
+
+      expect(errorsSpy.addError.calledOnce).to.be.true;
+      expect(errorsSpy.addError.calledWith('End date must be after start date'))
+        .to.be.true;
+    });
+
+    it('should use custom error messages', () => {
+      const customMessages = {
+        from: 'Custom from error',
+        to: 'Custom to error',
+        range: 'Custom range error',
+      };
+
+      validatePartialDateRange(
+        errorsSpy,
+        'invalid-from-date',
+        '2023-12-31',
+        customMessages,
+      );
+
+      expect(errorsSpy.addError.calledWith(customMessages.from)).to.be.true;
+    });
+
+    it('should not validate range for partial dates', () => {
+      // Partial dates should not have range validation applied
+      validatePartialDateRange(
+        errorsSpy,
+        '2023-XX-XX',
+        '2020-XX-XX', // Earlier year but should not error due to partial dates
+      );
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+
+    it('should handle date field objects', () => {
+      validatePartialDateRange(
+        errorsSpy,
+        { month: '1', day: '1', year: '2023' },
+        { month: '12', day: '31', year: '2023' },
+      );
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+
+    it('should handle mixed input types', () => {
+      validatePartialDateRange(errorsSpy, '2023-01-01', {
+        month: '12',
+        day: '31',
+        year: '2023',
+      });
+
+      expect(errorsSpy.addError.called).to.be.false;
+    });
+  });
+
+  describe('normalizeDateInput integration', () => {
```

Comment:
Did you intend to confirm direct testing of `normalizeDateInput` here via explicit calls in the test code?